Added "Draw fractal dispatching work to a thread pool" example

budziq opened the pull request:
fixes: https://github.com/brson/rust-cookbook/issues/102

Currently the example is marked as "no_run" as it leaves `output.png` in rust-cookbook cwd due to https://github.com/brson/rust-skeptic/issues/23 (there is a PR for it https://github.com/brson/rust-skeptic/pull/26)
Also on weaker machines the example in debug build might take 1-2s and I would not like to make the test even slower.

Also maybe we could add additional thumbnail for the output.png?
![output](https://cloud.githubusercontent.com/assets/221000/26546700/9be34e80-446b-11e7-81dc-dd9871614ea1.png)

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+
+[![threadpool-badge]][threadpool] [![num-badge]][num] [![num_cpus-badge]][num_cpus] [![image-badge]][image] [![cat-concurrency-badge]][cat-concurrency][![cat-science-badge]][cat-science][![cat-rendering-badge]][cat-rendering]
+
+Draws a fractal from [Julia set] to an image utilizing a thread pool for computation.
```
Add a thumbnail here with a link to the full sized image.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+
+// Function converting intensity values to RGB
+// Based on http://www.efg2.com/Lab/ScienceAndEngineering/Spectra.htm
+fn wavelength_to_rgb(wavelength: u32) -> Rgb<u8> {
```
Hide this like we hide all the error handling code. This is not important to understanding threadpool.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+}
+
+// Normalizes color intensity values within RGB range
+fn normalize(color: f32, factor: f32) -> u8 {
```
Hide this too.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+}
+
+// Maps Julia set distance estimation to intensity values
+fn julia(c: Complex<f32>, x: u32, y: u32, width: u32, height: u32, max_iter: u32) -> u32 {
```
Hide this too.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+
+    let iterations = 300;
+    // precomputed pixel colors from possible intensity range
+    let color_lut: Vec<_> = (0..iterations)
```
Remove this and inline the computation below. This seems like a performance optimization that has nothing to do with threadpool so it doesn't provide value to the example.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+        let pixel = color_lut[i as usize];
+        img.put_pixel(x, y, pixel);
+    }
+    let _ = img.save("output.png");
```
Show how to handle this error.

dtolnay commented on src/concurrency.md:
```diff
@@ -63,24 +64,170 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
+}
+
+fn run() -> Result<()> {
+    // resultant image size
```
Remove this comment. I think the names `width` and `height` and the next line are self-explanatory.

budziq commented:
Rebased and fixed :)

Don't know how I've missed that Result on `ImageBuffer::save`. Maby we should have `![deny(warnings)]` somewhere in skeptic config (in such way we could be also asked to fix snippets on depreciations)

Also I did not know a better way to handle the thumbnail except for inline html pointing to github static content.
